logics/impls/controller/mod.rs

```diff
@@ -1214,7 +1402,9 @@ impl<T: Storage<Data>> Internal for T {
         caller_pool: Option<(AccountId, PoolAttributes)>,
     ) -> Result<(U256, U256)> {
         // For each asset the account is in
-        let account_assets = self._account_assets(account, token_modify);
+        let markets = self._markets();
+        let caller = Self::env().caller();
+        let mut skip_pool = caller;
```

Comment:
How about declaring skip_pool as Option<AccountId> as follows?

```rust
let mut skip_pool = None;
```



Without caller_pool, both caller and skip_pool will contain Self::env().caller(), which makes the following branch a little difficult to understand

```rust
if *asset == caller || *asset == skip_pool {
    continue
}
```